integrations/rocket/src/lib.rs

```diff
@@ -119,6 +119,32 @@ impl Request {
     {
         Response(schema.execute(self.0).await.into())
     }
+
+    /// Specify the operation name of the request.
+    pub fn operation_name<T: Into<String>>(self, name: T) -> Self {
```

Comment:
What's the use for this? `operation_name` is set by the client when the request is deserialized, servers shouldn't have to change it.

Also, is there really a benefit to being able to write `.operation_name(...).0` over `.0.operation_name(...)`? You'll have to convert it to an `async_graphql::Request` anyway, so we might as well attempt to keep the API surface small.

Reply:
Yeah ok, will remove it

Reply:
Probably all the convenience wrappers since the same logic can be applied to them all.

Reply:
Ok, dropped commit 97b8ac3 entirely.

Reply:
Damn...i think leaving the `data()` function would have been nice. It is much nicer to write
```rust
#[rocket::post("/?<query..>", rank = 2)]
async fn graphql_query(schema: State<'_, StarWarsSchema>, db: Db, query: Request) -> Response {
    query.data(db).execute(&schema).await
}
```
instead of 😅
```rust
#[rocket::post("/?<query..>", rank = 2)]
async fn graphql_query(schema: State<'_, StarWarsSchema>, db: Db, mut query: Request) -> Response {
    query.0.data.insert(db);
    query.execute(&schema).await
}
```

Reply:
@Koxiaet What do you think about that? ☝